### PR TITLE
Fix to_rgb_color_list when passing rgba colors

### LINKED ISSUE #3477: create_annotated_heatmap throws on rgba in colorscale
```py
import plotly.figure_factory as ff

z = [
    [0.1, 0.3, 0.5, 0.7, 0.9],
    [1, 0.8, 0.6, 0.4, 0.2],
    [0.2, 0, 0.5, 0.7, 0.9],
    [0.9, 0.8, 0.4, 0.2, 0],
    [0.3, 0.4, 0.5, 0.7, 1],
]

# works
ff.create_annotated_heatmap(z, colorscale=[(0, "rgb(0, 0, 0)"), (1, "green")])
# throws error
ff.create_annotated_heatmap(z, colorscale=[(0, "rgba(0, 0, 0, 0)"), (1, "green")])
```

**Full stack trace**

```py
t.py in <module>
----> 62 ff.create_annotated_heatmap(
      63     z,
      64     colorscale=[(0, "rgba(0, 0, 0, 0)"), (1, "green")],
      65 )
      66 

plotly/figure_factory/_annotated_heatmap.py in create_annotated_heatmap(z, x, y, annotation_text, colorscale, font_colors, showscale, reversescale, **kwargs)
    105     colorscale = colorscale_validator.validate_coerce(colorscale)
    106 
--> 107     annotations = _AnnotatedHeatmap(
    108         z, x, y, annotation_text, colorscale, font_colors, reversescale, **kwargs
    109     ).make_annotations()

plotly/figure_factory/_annotated_heatmap.py in make_annotations(self)
    289             the heatmap
    290         """
--> 291         min_text_color, max_text_color = _AnnotatedHeatmap.get_text_color(self)
    292         annotations = []
    293         for n, row in enumerate(self.z):

plotly/figure_factory/_annotated_heatmap.py in get_text_color(self)
    261         elif isinstance(self.colorscale, list):
    262 
--> 263             min_col = to_rgb_color_list(self.colorscale[0][1], [255, 255, 255])
    264             max_col = to_rgb_color_list(self.colorscale[-1][1], [255, 255, 255])
    265 

plotly/figure_factory/_annotated_heatmap.py in to_rgb_color_list(color_str, default)
    149 def to_rgb_color_list(color_str, default):
    150     if "rgb" in color_str:
--> 151         return [int(v) for v in color_str.strip("rgb()").split(",")]
    152     elif "#" in color_str:
    153         return clrs.hex_to_rgb(color_str)

plotly/figure_factory/_annotated_heatmap.py in <listcomp>(.0)
    149 def to_rgb_color_list(color_str, default):
    150     if "rgb" in color_str:
--> 151         return [int(v) for v in color_str.strip("rgb()").split(",")]
    152     elif "#" in color_str:
    153         return clrs.hex_to_rgb(color_str)

ValueError: invalid literal for int() with base 10: 'a(0'
```

### PR DESCRIPTION
**Pull request by janosh**
Closes #3477.

## Code PR

- [x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ ] I have added tests (if submitting a new feature or correcting a bug) or modified existing tests. 
No tests for `to_rgb_color_list` at all yet but not necessary as dependent function `create_annotated_heatmap` is about to be deprecated (see below).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.

### CODE REVIEW

**Comment by janosh** [comment]
Another probable issue is that `to_rgb_color_list` does not handle `hsl(a)` colors at all, even though  `create_annotated_heatmap` does. Should we add a case for that?

**Comment by nicolaskruchten** [comment]
Hi, and thanks for your pull request! Yes, this probably needs a changelog entry.

I should note that in the next release of this library, we will be marking `create_annotated_heatmap` as deprecated, as `px.imshow()` will be able to add text to heatmaps natively, and all the colour-handling will happen in the Javascript layer so as to be more robust.

**Comment by janosh** [comment]
Good to know. So does `px.imshow` on current master already handle text? If so I might switch right away.

**Comment by janosh** [comment]
@nicolaskruchten This is ready for review.

**Comment by nicolaskruchten** [comment]
Thanks @janosh ! Yes, I've just merged the new changes to `px.imshow` to master, and I've marked `create_annotated_heatmap` as deprecated, so I'd rather not make any new modifications to it now, if that's OK.

**Comment by janosh** [comment]
It's still a fix and would allow me to release some code that relies on this. Merging fixes even to deprecated code makes sense imo.

File: packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py [hunk 1]
```diff
@@ -147,9 +147,10 @@ def create_annotated_heatmap(
 
 
 def to_rgb_color_list(color_str, default):
-    if "rgb" in color_str:
-        return [int(v) for v in color_str.strip("rgb()").split(",")]
-    elif "#" in color_str:
+    color_str = color_str.strip()
+    if color_str.startswith("rgb"):
+        return [int(v) for v in color_str.strip("rgba()").split(",")]
```
Comment (nicolaskruchten) [inline packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py:152]: does this now still work with `rbg(...)` and `rgba(...)` both ?

Comment (janosh) [reply packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py:152]: Yes, it does. Tested it. As I mentioned, doesn't handle HSL(A) yet.

Comment (nicolaskruchten) [reply packages/python/plotly/plotly/figure_factory/_annotated_heatmap.py:152]: OK. After this next release I won't be accepting any more PRs to this deprecated code so let's not work on any HSL stuff please :)

**Comment by nicolaskruchten** [comment]
OK, thanks!

**Comment by janosh** [comment]
Thanks for merging!

**Comment by janosh** [comment]
> I should note that in the next release of this library, we will be marking `create_annotated_heatmap` as deprecated, as `px.imshow()` will be able to add text to heatmaps natively, and all the colour-handling will happen in the Javascript layer so as to be more robust.

`px.imshow` doesn't seem to provide a way to set `xgap`/`ygap`. Also would be cool if the https://plotly.com/python/imshow examples included one with pandas where a df with a text column is used to render hover tooltips. Doesn't feel like `imshow` is ready to replace `create_annotated_heatmap` yet.

**Pull request merged**